src/main/java/de/tum/in/www1/artemis/service/LearningGoalService.java

```diff
@@ -0,0 +1,115 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.LearningGoal;
+import de.tum.in.www1.artemis.domain.Result;
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.lecture.ExerciseUnit;
+import de.tum.in.www1.artemis.domain.lecture.LectureUnit;
+import de.tum.in.www1.artemis.domain.participation.StudentParticipation;
+import de.tum.in.www1.artemis.repository.LearningGoalRepository;
+import de.tum.in.www1.artemis.repository.ResultRepository;
+import de.tum.in.www1.artemis.web.rest.dto.LearningGoalProgress;
+
+@Service
+public class LearningGoalService {
+
+    private final LearningGoalRepository learningGoalRepository;
+
+    private final AuthorizationCheckService authorizationCheckService;
+
+    private final ResultRepository resultRepository;
+
+    private final ParticipationService participationService;
+
+    public LearningGoalService(LearningGoalRepository learningGoalRepository, AuthorizationCheckService authorizationCheckService, ParticipationService participationService,
+            ResultRepository resultRepository) {
+        this.learningGoalRepository = learningGoalRepository;
+        this.authorizationCheckService = authorizationCheckService;
+        this.participationService = participationService;
+        this.resultRepository = resultRepository;
+    }
+
+    /**
+     * Calculate the progress in a learning goal for a specific user
+     * @param learningGoal learning goal to get the progress for
+     * @param user user to get the progress for
+     * @return progress of the user in the learning goal
+     */
+    public LearningGoalProgress getLearningGoalPerformance(LearningGoal learningGoal, User user) {
```

Comment:
```suggestion
    public LearningGoalProgress calculateLearningGoalProgress(LearningGoal learningGoal, User user) {
```

Reply:
Done in da86f2c21800028e9a9a960cd727cf04ede905c7